auth-foundation/src/main/java/com/okta/authfoundation/GrantType.kt

```diff
@@ -0,0 +1,125 @@
+     * owner has a trust relationship with the client.
+     *
+     * @see [RFC 6749, Section 4.3](https://tools.ietf.org/html/rfc6749#section-4.3)
+     */
+    object Password : GrantType("password")
+
+    /**
+     * The device authorization grant is used by browserless or input-constrained devices
+     * in the device flow to obtain an access token.
+     *
+     * @see [RFC 8628](https://tools.ietf.org/html/rfc8628)
+     */
+    object DeviceCode : GrantType("urn:ietf:params:oauth:grant-type:device_code")
+
+    /**
+     * A grant type for exchanging a token of one type for a token of another type.
+     *
+     * @see [RFC 8693](https://tools.ietf.org/html/rfc8693)
+     */
+    object TokenExchange : GrantType("urn:ietf:params:oauth:grant-type:token-exchange")
+
+    /**
+     * A grant type that uses a JWT as an authorization grant.
+     *
+     * @see [RFC 7523](https://tools.ietf.org/html/rfc7523)
+     */
+    object JwtBearer : GrantType("urn:ietf:params:oauth:grant-type:jwt-bearer")
+
+    /**
+     * A grant type for One-Time Passcode (OTP) authentication, often used as a second factor.
+     *
+     * @see [Okta Direct Authentication OTP](https://developer.okta.com/docs/reference/direct-auth/grant-types/#otp)
+     */
+    object Otp : GrantType("urn:okta:params:oauth:grant-type:otp")
+
+    /**
+     * A grant type for Multi-Factor Authentication (MFA) using One-Time Passcodes (OTP).
+     *
+     * @see [Okta MFA OTP](https://developer.okta.com/docs/guides/configure-direct-auth-grants/bmfaotp/main/#about-the-direct-authentication-grant)
+     */
+    object OtpMfa : GrantType("http://auth0.com/oauth/grant-type/mfa-otp")
+
+    /**
+     * A grant type for Out-of-Band (OOB) authentication, such as push notifications.
+     *
+     * @see [Okta Direct Authentication OOB](https://developer.okta.com/docs/reference/direct-auth/grant-types/#oob)
+     */
+    object Oob : GrantType("urn:okta:params:oauth:grant-type:oob")
+
+    /**
+     * A grant type for Multi-Factor Authentication (MFA) using Out-of-Band (OOB) methods.
+     *
+     * @see [Okta MFA OOB](https://developer.okta.com/docs/guides/configure-direct-auth-grants/dmfaoobov/main/#direct-authentication-mfa-oob-okta-verify-push-flow)
+     */
+    object OobMfa : GrantType("http://auth0.com/oauth/grant-type/mfa-oob")
+
+    /**
+     * A grant type for WebAuthn authentication.
+     *
+     * @see [Okta Direct Authentication WebAuthn](https://developer.okta.com/docs/reference/direct-auth/grant-types/#webauthn)
+     */
+    object WebAuthn : GrantType("urn:okta:params:oauth:grant-type:webauthn")
+
+    /**
+     * A grant type for Multi-Factor Authentication (MFA) using WebAuthn.
+     *
+     * @see [Okta MFA WebAuthn](https://developer.okta.com/docs/guides/configure-direct-auth-grants/bmfawebauthn/main/#about-the-direct-authentication-grant)
+     */
+    object WebAuthnMfa : GrantType("urn:okta:params:oauth:grant-type:mfa-webauthn")
```

Comment:
I see a lot of repeated MFA/non-MFA types. Should we consider MFA a field in data class for such cases?

Reply:
MFA are different grant types. they are used in different context. Adding them to one would require logic to differentiate the context. for example 

`data class WebAuthn(isMfa: Boolean)`

The usage and documentation will then need to check the context of isMfa for every usage of WebAuthn. It is clearer to specify it as a type.